src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewDataEvaluator.cs

```diff
@@ -205,17 +207,5 @@ private static ViewDataInfo GetPropertyValue(object container, string propertyNa
 
             return new ViewDataInfo(container, propertyInfo, () => propertyInfo.GetValue(container));
```

Comment:
Yes, I was thinking about the problem from the wrong angle. Think #5325 is a start down the road.